programs/uxd/src/utils/math/compute_amount_fraction.rs

```diff
@@ -0,0 +1,21 @@
+use crate::error::UxdError;
+use crate::utils::checked_convert_u128_to_u64;
+use anchor_lang::prelude::*;
+
+// Precision loss may lower the returned value amount.
+// Precision loss of 1 native unit may be expected.
+pub fn compute_amount_fraction(
```

Comment:
this is similar to `compute_amount_less_fraction`, but the precision loss is in the oposite direction (which is why we cant use `compute_amount_less_fraction` instead